src/core/qgsfield.cpp

```diff
@@ -258,6 +259,22 @@ QString QgsField::displayString( const QVariant &v ) const
     return QgsApplication::nullRepresentation();
   }
 
+  if ( v.userType() == QMetaType::type( "QgsReferencedGeometry" ) )
+  {
+    QgsReferencedGeometry geom = qvariant_cast<QgsReferencedGeometry>( v );
+    if ( geom.isNull() )
+      return QgsApplication::nullRepresentation();
+    else
+    {
+      QString formattedText = QStringLiteral( "%1 [%2]" ).arg( geom.asWkt(), geom.crs().userFriendlyIdentifier() );
+      if ( formattedText.length() >= 1000 )
+      {
+        formattedText = formattedText.left( 999 ) + QChar( 0x2026 );
```

Comment:
```suggestion
      if ( formattedText.length() >= 1050 )
      {
        formattedText = formattedText.left( 999 ) + QChar( 0x2026 );
```
I'd use a limit higher than what is actually use as truncation.
It's a bit stupid to truncate something at 1000 if it's 1005.